bindings/rust/extended/s2n-tls/src/callbacks/cert_validation.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls_sys::*;
+
+use crate::{
+    connection::Connection,
+    error::{Error, Fallible},
+};
+use std::{marker::PhantomData, ptr::NonNull};
+
+pub struct CertValidationInfo<'a> {
+    info: NonNull<s2n_cert_validation_info>,
+    _lifetime: PhantomData<&'a s2n_cert_chain_and_key>,
+}
+
+impl CertValidationInfo<'_> {
+    pub(crate) fn from_ptr(info: *mut s2n_cert_validation_info) -> Self {
+        let info = NonNull::new(info).expect("info pointer is not null");
+        CertValidationInfo {
+            info,
+            _lifetime: PhantomData,
+        }
+    }
+
+    pub(crate) fn as_ptr(&mut self) -> *mut s2n_cert_validation_info {
+        self.info.as_ptr()
+    }
+
+    /// Corresponds to [s2n_cert_validation_accept].
+    pub fn accept(&mut self) -> Result<(), Error> {
```

Comment:
`accept`/`reject` probably shouldn't be public if we are calling these on the user's behalf.